test/tbb/test_concurrent_queue.cpp

```diff
@@ -281,3 +282,211 @@ TEST_CASE("Test clear and dtor with TrackableItem") {
+    REQUIRE_MESSAGE(src.get_allocator() == dst.get_allocator(), "Incorrect test setup: allocators should be equal");
+    REQUIRE_MESSAGE(cpy.unsafe_size() == dst.unsafe_size(), "Queues are not equal");
+
+    test_check_move_allocator<tbb::concurrent_bounded_queue<std::vector<int>>>(src_bnd, dst_bnd, cpy_bnd);
+    REQUIRE_MESSAGE(src.get_allocator() == dst.get_allocator(), "Incorrect test setup: allocators should be equal");
+    REQUIRE_MESSAGE(cpy_bnd.size() == dst_bnd.size(), "Queues are not equal");
+}
+
+void test_move_assignment_test_unequal() {
+    stateful_allocator<int> src_alloc;
+    std::vector<int, stateful_allocator<int>> v(8, 0, src_alloc);
+    tbb::concurrent_queue<std::vector<int, stateful_allocator<int>>, stateful_allocator<int>> src(src_alloc);
+
+    src_alloc.state = 0;
+    v.push_back(42);
+    v.push_back(82);
+    src.push(v);
+    src.push(v);
+
+    stateful_allocator<int> dst_alloc;
+    dst_alloc.state = 1;
+    tbb::concurrent_queue<std::vector<int, stateful_allocator<int>>, stateful_allocator<int>> dst(dst_alloc);
+    tbb::concurrent_queue<std::vector<int, stateful_allocator<int>>, stateful_allocator<int>> cpy(src_alloc);
+    cpy = src;
+    dst = std::move(src);
+
+    tbb::concurrent_bounded_queue<std::vector<int, stateful_allocator<int>>, stateful_allocator<int>> src_bnd(src_alloc);
+    tbb::concurrent_bounded_queue<std::vector<int, stateful_allocator<int>>, stateful_allocator<int>> dst_bnd(dst_alloc);
+    tbb::concurrent_bounded_queue<std::vector<int, stateful_allocator<int>>, stateful_allocator<int>> cpy_bnd(src_alloc);
+    src_bnd.push(v);
+    src_bnd.push(v);
+    cpy_bnd = src_bnd;
+    dst_bnd = std::move(src_bnd);
+
+    test_check_move_allocator<tbb::concurrent_queue<std::vector<int, stateful_allocator<int>>, stateful_allocator<int>>>(src, dst, cpy);
+    REQUIRE_MESSAGE(src.get_allocator() != dst.get_allocator(), "Incorrect test setup: allocators should be unequal");
+    REQUIRE_MESSAGE(src.unsafe_size() == 0, "Moved from container should not contain any elements");
+    REQUIRE_MESSAGE(dst.unsafe_size() == cpy.unsafe_size(), "Queues are not equal");
+    REQUIRE_MESSAGE(std::equal(dst.unsafe_begin(), dst.unsafe_end(), cpy.unsafe_begin()), "Elements are not equal");
+
+    test_check_move_allocator<tbb::concurrent_bounded_queue<std::vector<int, stateful_allocator<int>>, stateful_allocator<int>>>(src_bnd, dst_bnd, cpy_bnd);
+    REQUIRE_MESSAGE(src.get_allocator() != dst.get_allocator(), "Incorrect test setup: allocators should be unequal");
+    REQUIRE_MESSAGE(src_bnd.size() == 0, "Moved from container should not contain any elements");
+    REQUIRE_MESSAGE(dst_bnd.size() == cpy_bnd.size(), "Queues are not equal");
+    REQUIRE_MESSAGE(std::equal(dst_bnd.unsafe_begin(), dst_bnd.unsafe_end(), cpy_bnd.unsafe_begin()), "Elements are not equal");
+}
+
+//! move assignment test for equal and unequal allocator
+//! \brief \ref error_guessing
+TEST_CASE("concurrent_queue") {
```

Comment:
Please provide the better name for test case

Reply:
renamed